test/graphql-api/utils/executeOperation.ts

```diff
@@ -83,3 +66,59 @@ export const executeOperation = async (
     return page.waitForResponse(urlString).then(resolve)
   })
 }
+
+/**
+ * Executes a GraphQL operation in the given Puppeteer context.
+ */
+export const executeOperation = async (
+  page: Page,
+  payload: GraphQLRequestPayload,
+  options?: GraphQLOperationOptions,
+) => {
+  const { uri = HOSTNAME, method = 'POST' } = options || {}
+  const { query, variables } = payload
+  const url = new URL(uri)
+
+  if (method === 'GET') {
+    url.searchParams.set('query', query)
+
+    if (variables) {
+      url.searchParams.set('variables', JSON.stringify(variables))
+    }
+  }
+
+  const urlString = url.toString()
+
+  return performRequest({
+    urlString,
+    method,
+    payload: { query, variables },
+    page,
+  })
+}
+
+/**
+ * Executes a GraphQL batch operation in the given Puppeteer context.
+ */
+export const executeBatchOperation = async (
+  page: Page,
+  payload: GraphQLRequestPayload[],
+  options?: Override<GraphQLOperationOptions, { method: 'POST' }>,
+) => {
+  const { uri = HOSTNAME, method = 'POST' } = options || {}
+  const url = new URL(uri)
+  const urlString = url.toString()
+
+  const results = await Promise.all(
+    payload.map(({ query, variables }) =>
+      performRequest({
+        page,
+        urlString,
```

Comment:
As these requests will be to the same URL, `performRequest` will resolve on the first match. This results in both results having the same JSON body.